src/core/link.lua

```diff
@@ -2,11 +2,16 @@ module(...,package.seeall)
 
 local debug = _G.developer_debug
 
+local shm = require("core.shm")
 local ffi = require("ffi")
 local C = ffi.C
 
 local packet = require("core.packet")
 require("core.packet_h")
+
+local counter = require("core.counter")
+require("core.counter_h")
```

Comment:
this is redundant, since `core.counter` already defines the `struct counter`.  In fact, the `require()` function won't do anything because `package.loaded.counter_h` is already non-nil.